src/docs/tips.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Advanced Tips
+---
+
+# Advanced Tips
+
+In this section we'll outline some advanced hints and tips to get the most out of developing with Eclipse Theia.
+
+## Upgrading functionality in a VS Code Extension for use in Theia
+
+It's possible to declare a custom Theia API namespace which can be implemented in a standard VS Code Extension.
+In this way, common functionality can be written to target VS Code and Theia, with enhancements appearing just in the Theia IDE.
+
+Usage of the custom API is invoked only when run in a Theia IDE, determined via the application name.
+The API is imported asynchronously to avoid runtime errors in VS Code.
+
+```typescript
+if (vscode.env.appName === APP_NAME) {
+    // Implement Theia API
+    const api = await import('@extension/api');
+    api.host.getMessageHandler(() => getMessage());
```

Comment:
```suggestion
    // After importing the custom API, it can be used as any other API. The following lines are using an example API.
    api.host.getMessageHandler(() => getMessage());
```